### ISSUE TITLE
IP-Adapter-FaceID with SDXL model?

### ISSUE THREAD

**Issue by dawei03896** [tencent-ailab/IP-Adapter#314]

**Comment by xiaohu2015** [comment]
https://huggingface.co/h94/IP-Adapter-FaceID/blob/main/ip-adapter-faceid-portrait_sdxl.bin

**Comment by xddun** [comment]
Can you provide an example of using this ip-adapter-faceid-portrait_sdxl.bin? The example I wrote is causing an error.

**Comment by xiaohu2015** [comment]
it is same as sd15 faceid-portrait, you can use https://github.com/cubiq/ComfyUI_IPAdapter_plus

**Comment by xddun** [comment]
Indeed, as you said, I have successfully conducted the experiment. Thank you.

 I also want to know how to train the ip-adapter-faceid-portrait_sdxl.bin. Could you provide some suggestions? Thanks.

**Comment by xiaohu2015** [comment]
you can refer to https://github.com/tencent-ailab/IP-Adapter/wiki/IP%E2%80%90Adapter%E2%80%90Face

for portrait, we use portrait data to train, we use face bounding box to crop portrait data （with 2x region）. For SDXL, we train it at 1024x1024

**Comment by xddun** [comment]
I have currently thoroughly experienced this project, and I feel that the model structure is truly delightful. You must have gone through many experimental paths to achieve such good results. IP-Adapter-FaceID, especially this version: ip-adapter-faceid-portrait_sdxl.bin, exhibits great flexibility in facial similarity. Compared to InstantID, it's much more flexible! This is a very good solution.

I currently have several questions and would appreciate your help in clearing my doubts:

1. Is IP-Adapter-FaceID supposed to be adapted to the SD model during training? Since the underlying noise reduction layers of SD's Unet rely on CLIP, the IP-Adapter-FaceID you trained can be used with other SD models, but isn't it more suitable for other SD models if the IP-Adapter-FaceID model is fine-tuned alongside them (for finer results)? I believe this is the case, am I correct?

2. In comparative analysis, I distinctly feel that the characters produced by the ip-adapter-faceid-portrait_sdxl.bin model are better and more similar to the input individuals. However, I also sadly realize that the training code probably requires more professional expertise to use. I hope this aspect can be improved upon to better leverage the advantages of IP-Adapter-FaceID. I hope to receive a more detailed training process for ip-adapter-faceid-portrait_sdxl.bin.

3. How much data is required to train ip-adapter-faceid-portrait_sdxl.bin?

4. Does the input for ip-adapter-faceid-portrait_sdxl.bin consist of embedding features of multiple facial images, with the output being the selection of a particular input image? I believe that such an approach might lead to training failure. Therefore, I hope more information can be disclosed regarding the process here.

**Comment by xddun** [comment]
5. Have you considered a way to control the proportions of the faces, such as incorporating landmarks like in InstantID?

**Comment by xiaohu2015** [comment]
@xddun hi, the training of faceid-portrait is almost the same as that of faceid, except for the following two differences:
(1) faceid-portrait don't use lora module;
(2) faceid-portrait uses portrait images (we firstly use face detector to get face region, the we expand the image to 2x of the region) to train.

**Comment by xiaohu2015** [comment]
> 3\. How much data is required to train ip-adapter-faceid-portrait_sdxl.bin?

1) all the IP-Adapter models are trained with base SD models (e.g.  original SD1.5 and SDXL), once trained, it can be used for other finetuned SD models.
2) you can refer to this https://github.com/tencent-ailab/IP-Adapter/blob/main/tutorial_train_faceid.py, and remove LoRA, and use face detector to crop portrait images.
3) about 60w images;
4) no, in the training stage, it just use one face embedding;
5) I think it can, but I have not try that.

**Comment by xddun** [comment]
Thank you for your response, they are all very good experiences!

**Comment by katarzynasornat** [comment]
> you can refer to https://github.com/tencent-ailab/IP-Adapter/wiki/IP%E2%80%90Adapter%E2%80%90Face
> 
> for portrait, we use portrait data to train, we use face bounding box to crop portrait data （with 2x region）. For SDXL, we train it at 1024x1024

Is there a support for SDXL when it comes to ip-adapter-faceid? When I use it, am I able to generate full scene with just face fixed? Cause I tried to use p-adapter-faceid and I got just kind of portrait of image. (e.g my prompt was "a girl ridding a horse, full pose" - then the horse and a girl was cutted and when I generated without faceid adapter I got normal full pose).

thanks for help

**Comment by YIYANGCAI** [comment]
> Thank you for your response, they are all very good experiences!

Hi, how can you initialize the ip-adapter with this checkpoint ip-adapter-faceid-portrait_sdxl.bin? I use IPAdapterFaceIDXL to do the initialization:
```python
import torch
from diffusers import StableDiffusionXLPipeline, DDIMScheduler
from PIL import Image

from ip_adapter.ip_adapter_faceid import IPAdapterFaceIDXL
base_model_path = "SG161222/RealVisXL_V3.0"
ip_ckpt = "ip-adapter-faceid-portrait_sdxl.bin"
device = "cuda"

noise_scheduler = DDIMScheduler(
    num_train_timesteps=1000,
    beta_start=0.00085,
    beta_end=0.012,
    beta_schedule="scaled_linear",
    clip_sample=False,
    set_alpha_to_one=False,
    steps_offset=1,
)
pipe = StableDiffusionXLPipeline.from_pretrained(
    base_model_path,
    torch_dtype=torch.float16,
    scheduler=noise_scheduler,
    add_watermarker=False,
)

# load ip-adapter
ip_model = IPAdapterFaceIDXL(pipe, ip_ckpt, device)
```
However, I got this error, seems that this checkpoint does not match with the model class. 
```
RuntimeError: Error(s) in loading state_dict for MLPProjModel:
        size mismatch for proj.2.weight: copying a param with shape torch.Size([32768, 1024]) from checkpoint, the shape in current model is torch.Size([8192, 1024]).
        size mismatch for proj.2.bias: copying a param with shape torch.Size([32768]) from checkpoint, the shape in current model is torch.Size([8192]).
```
Could you give a clue about using this checkpoint? Thank you!

**Comment by jadechip** [comment]
Stuck with the same issue
